### DISCUSSION TITLE
Markdown videos with Reveal JS or Slide.dev

### DISCUSSION THREAD

**Discussion by LiveDuo** [remotion-dev/remotion#2448] [General]
### Background

A significant amount of educational videos are still made with powerpoint, even after 15+ years of Youtube and online videos. That's probably because __ it's cumbersome with specialised tools such as [3b1b/manim](https://github.com/3b1b/manim) and for many creators it's not worth to spend the time learning something like Adobe Premier (and the money) given that more educational videos are just text and images with voice overs.

### Markdown in Remotion

An intermediate solution could be to use a structured format such as markdown to describe most of the HTML of a video while using a tool like Remotion to help with rendering. There are many tools that can do the markdown part as [Reveal Js](https://revealjs.com) and [Slide Dev](https://sli.dev/). For more info, you might want to glance at this video [Markdown Presentation Tools](https://www.youtube.com/watch?v=owx5KoiqFBs) *.

In essence, there should be two components for this to work: 
1. The markdown to HTML part that will be used to speed up the HTML design
2. And the Remotion part to help with previewing the HTML and export to video

*I'm not associated with its creator

### What's next

I'll probably experiment with Remotion and Reveal / Slide Dev in the next few weeks and I'll be interested to hear what others think.

Do you find this interesting? Anyone got a similar thing to work?

**Comment by LiveDuo** [comment]
Update: I've been playing with `react-markdown` and it was easier than I initially thought. It's just a very barebones MVP I put together last night and is missing a lot. Here's the [source](https://github.com/LiveDuo/remotion-markdown-example/blob/master/src/Composition.tsx).

**Comment by JonnyBurger** [comment]
Very nice idea!
Maybe you also find MDX useful: https://www.remotion.dev/docs/webpack#enabling-mdx-support (page is 2 years old, hope it still works like that)

With a MDX loader, you can import an .md file and get syntax highlighting. Even more powerful, you can embed React components in your Markdown if you need to escape the limitations of Markdown!

**Reply by LiveDuo** [reply]
Thanks!

> With a MDX loader, you can import an .md file and get syntax highlighting. Even more powerful, you can embed React components in your Markdown if you need to escape the limitations of Markdown!

I will try it out.

At the moment I'm thinking about animations, similar to [Powerpoint transitions](https://support.microsoft.com/en-us/office/add-change-or-remove-transitions-between-slides-3f8244bf-f893-4efd-a7eb-3a4845c9c971) and found [Remotion Animated](https://www.remotion-animated.dev/).

Thinking about one transition per slide (or two, one entry and out exit transitions). Any thoughts on the animation library? Or other options worth checking out?
